src/backend/src/main/java/com/linkedpipes/lpa/backend/ExecutionController.java

```diff
@@ -0,0 +1,45 @@
+package com.linkedpipes.lpa.backend;
+
+import com.linkedpipes.lpa.backend.services.HttpUrlConnector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.io.IOException;
+
+@RestController
+public class ExecutionController {
+
+    private static final Logger logger =
+            LoggerFactory.getLogger(DiscoveryController.class);
+
+    private HttpUrlConnector httpUrlConnector = new HttpUrlConnector();
+
+    @RequestMapping("/execution/status")
+    public ResponseEntity<String> getStatus(@RequestParam( value="executionIri") String executionIri) throws IOException{
+        if(executionIri == null || executionIri.isEmpty()) {
+            return new ResponseEntity("Execution IRI not provided.", HttpStatus.BAD_REQUEST);
+        }
+
+        String response = httpUrlConnector.sendGetRequest(executionIri + "/overview",
+                null, "application/json");
+
+        return ResponseEntity.ok(response);
+    }
+
+    @RequestMapping("/execution/result")
+    @ResponseBody
+    public ResponseEntity<String> getResult(@RequestParam( value="executionIri") String executionIri) throws IOException{
+        if(executionIri == null || executionIri.isEmpty()) {
+            return new ResponseEntity("Execution IRI not provided.", HttpStatus.BAD_REQUEST);
+        }
+
+        String response = httpUrlConnector.sendGetRequest(executionIri,
+                null, "application/json");
+
+        return ResponseEntity.ok(response);
```

Comment:
Could be just me wondering but does this mean that I can get a string 
"Execution IRI not provided." as a body of response and another time some JSON object?

If so why not use JSON in both cases to make response processing easier?

Reply:
Very good point. From my experience:

In case of
- expected and handled business error: the API should return HTTP 200 with JSON object.
- handled error of type entity not found: the API should return HTTP 404
- expected and handled input error (e.g. wrong parameters): the API should return HTTP 400
- unexpected and unhandled error on server: the API should return HTTP 500 with error serialized in JSON
- always return JSON object, never JSON array or empty response. Returning object makes it futureproof, i.e. backwards compatible once you need to extend the response.

Reply:
Agreed, was planning to update that to return an error object instead. Will update accordingly